lti_consumer/lti_xblock.py

```diff
@@ -846,9 +848,8 @@ def resource_link_id(self):
         i4x-2-3-lti-31de800015cf4afb973356dbe81496df this part of resource_link_id:
         makes resource_link_id to be unique among courses inside same system.
         """
-        return str(urllib.parse.quote(
-            f"{self.runtime.hostname}-{self.location.html_id()}"  # pylint: disable=no-member
-        ))
+        hostname = self.runtime.service(self, 'hostname').get_hostname()
+        return str(urllib.parse.quote(f"{hostname}-{self.location.html_id()}"))
```

Comment:
I forgot to mention this, but `self.location` is just [a wrapper](https://github.com/open-craft/edx-platform/blob/4271e24eb90cd2e113eaa2250d807ff67e77a99d/xmodule/modulestore/xml_importer.py#L104-L107) for `self.scope_ids.usage_id`. Could we replace its usage in this XBlock?